src/ray/raylet/node_manager.cc

```diff
@@ -1600,9 +1726,48 @@ void NodeManager::HandleRequestWorkerLease(const rpc::RequestWorkerLeaseRequest
             reply->mutable_worker_address()->set_port(worker->Port());
             reply->mutable_worker_address()->set_worker_id(worker->WorkerId().Binary());
             reply->mutable_worker_address()->set_raylet_id(self_node_id_.Binary());
+            RAY_LOG(WARNING) << "leased_worker 2 = " << worker->WorkerId();
             RAY_CHECK(leased_workers_.find(worker->WorkerId()) == leased_workers_.end());
             leased_workers_[worker->WorkerId()] = worker;
-            leased_worker_resources_[worker->WorkerId()] = request_resources;
+// TODO (Ion): Fix handling floating point errors, maybe by moving to integers.               
+#define ZERO_CAPACITY 1.0e-5
+            TaskResourceInstances allocated_resources;
+            if (task_specs.IsActorCreationTask()) {
+              allocated_resources = worker->GetLifetimeAllocatedInstances();                                                   
+            } else {
+              allocated_resources = worker->GetAllocatedInstances();                                                   
+            }
+            auto predefined_resources = allocated_resources.predefined_resources;
+            for (size_t res_idx = 0; res_idx < predefined_resources.size(); res_idx++) {
```

Comment:
All of the protobuf formatting seems like it could be a helper function that takes in the predefined resources and the reply.